### PR TITLE
feat: add jina embedding

### PR DESCRIPTION
**Pull request by numb3r3**
Add support for Jina Inference API

- [x] `JinaEmbedding`, unittest, and docs
- [x] jupyter example
- [ ] raise PR in origin langchain repo

### CODE REVIEW

File: docs/ecosystem/jina.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Jina
+
+This page covers how to use the Jina ecosystem within LangChain.
+It is broken into two parts: installation and setup, and then references to specific Jina wrappers.
+
+## Installation and Setup
+- Install the Python SDK with `pip install jina`
+- Get a Jina Cloud auth token and set it as an environment variable (`JINA_AUTH_TOKEN`)
```
Comment (saahil) [inline docs/ecosystem/jina.md]: Can we link to Jina AI Cloud for token generation?

Also, please call it `Jina AI Cloud`.

File: docs/modules/indexes/examples/embeddings.ipynb [hunk 2]
```diff
@@ -268,6 +268,153 @@
...
+    "collapsed": false,
+    "pycharm": {
+     "is_executing": true
+    }
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "In the above example, we used the default Embedding model, `ViT-B-32::openai`. Let's load a different model (e.g., `ViT-B-32::laion2b-s34b-b79k`). For a full list of models, see [here](https://cloud.jina.ai/user/inference/model/63dca9df5a0da83009d519cd).\n"
```
Comment (saahil) [inline docs/modules/indexes/examples/embeddings.ipynb]: I feel that we don't need two completely different code examples for different models - I would recommend a two parameter example only, and below that cell, we can write
```
In the above example, `ViT-B-32::openai`, OpenAI's 32-bit CLIP model. For a full list of models, see [here](https://cloud.jina.ai/user/inference/model/63dca9df5a0da83009d519cd).\
```

File: docs/modules/indexes/examples/embeddings.ipynb [hunk 3]
```diff
@@ -268,6 +268,153 @@
...
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
```
Comment (saahil) [inline docs/modules/indexes/examples/embeddings.ipynb]: Rememeber to set the `execution_count` to `null` everywhere.

File: docs/ecosystem/jina.md (hunk 1 above)
Comment (numb3r3) [reply docs/ecosystem/jina.md]: @saahil reserved, please review the updates.

**Pull request merged**